elytron/common/src/main/java/org/wildfly/extension/elytron/CertificateAuthorityDefinition.java

```diff
@@ -92,15 +90,23 @@ public void validateParameter(String parameterName, ModelNode value) throws Oper
     }
 
     private static final AttributeDefinition[] ATTRIBUTES = new AttributeDefinition[]{URL, STAGING_URL};
-    private static final AbstractAddStepHandler ADD = new CertificateAuthorityAddHandler();
+    private static final ElytronReloadRequiredWriteAttributeHandler WRITE = new ElytronReloadRequiredWriteAttributeHandler(ATTRIBUTES);
 
-    private static final OperationStepHandler REMOVE = new TrivialCapabilityServiceRemoveHandler(ADD, CERTIFICATE_AUTHORITY_RUNTIME_CAPABILITY);
-    private static final AbstractWriteAttributeHandler WRITE = new ElytronReloadRequiredWriteAttributeHandler(ATTRIBUTES);
+    static CertificateAuthorityDefinition configure(final String subsystemName, final Class<?> extensionClass) {
```

Comment:
Since the `ElytronExtension` class contains the `SUBSYSTEM_NAME`, just wondering if we can make use of that directly instead of passing the subsystem name as a parameter here and in other places. Perhaps we could introduce a common Elytron extension class with a `getSubsystemName()` method and then the Elytron and Elytron TLS extensions could extend the common class and implement the `Extension` interface?

Reply:
Yeah this might work. Right now the methods that were pulled from ElytronExtension are also in `ElytronCommonDefinitions`, so I can try moving it to a new `ElytronCommonExtension`

Reply:
Did some testing, the main issue is that since the method `ElytronCommonExtension.getSubsystemName()` would be abstract, it can't be static, which also applies to anything using the subsystem name (like `configure()`). I can't think of another straightforward way to require that the `getSubsystemName()` method be overridden.